osf/models/preprint.py

```diff
@@ -380,7 +380,7 @@ def create_version(cls, create_from_guid, auth):
 
         # Create an entry in the `GuidVersionsThrough` table
         guid_version = GuidVersionsThrough(
-            referent=guid_obj.referent,
+            referent=preprint,
```

Comment:
Ah, I see, looks like it's from my refactor 🐛 . Maybe I moved some `if` checks around so it revealed this bug where `guid_obj.referent` is not changed for pre/post moderation during creation.